ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/product/v1/product/giftcardProduct.html

```diff
@@ -0,0 +1,126 @@
+<!--/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  ~
+  ~ Copyright 2019 Adobe. All rights reserved.
+  ~ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License. You may obtain a copy
+  ~ of the License at http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software distributed under
+  ~ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  ~ OF ANY KIND, either express or implied. See the License for the specific language
+  ~ governing permissions and limitations under the License.
+  ~
+  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/-->
+
+  <template data-sly-template.giftcardProduct="${ @ product, allowZeroQuantity, defaultQuantity}">
+   <fieldset class="giftcardProduct__amount">
+      <div id="giftcard-amount-box" class="giftcardProduct__field">
+         <label for="giftcard-amount-input" id="amount-label-input" class="giftcardProduct__label">
+            <span>Amount</span><span class="required">*</span>
+            <input type="hidden" id="giftcard_amount_input_uid" name="giftcard_amount_input_uid" value='${product["giftCardAttributes"]["giftCardOptions"]["Custom Giftcard Amount"]["value"]["uid"]}'>
+            <input data-sly-test='${product["giftCardAttributes"]["allowOpenAmount"]}' type="hidden" id="giftcard_amount_input_min" name="giftcard_amount_input_min" value='${product["giftCardAttributes"]["openAmountRange"]["openAmountMin"]}'>
+            <input data-sly-test='${product["giftCardAttributes"]["allowOpenAmount"]}' type="hidden" id="giftcard_amount_input_max" name="giftcard_amount_input_max" value='${product["giftCardAttributes"]["openAmountRange"]["openAmountMax"]}'>
+         </label>
+         <div>
+            <input type="text" id="giftcard_amount_input" name="giftcard_amount_input">
+            <p class="note">
+               <span data-sly-test='${product["giftCardAttributes"]["allowOpenAmount"]}'>Minimum: ${product["giftCardAttributes"]["openAmountRange"]["openAmountMin"]} </span> <span data-sly-test='${product["giftCardAttributes"]["allowOpenAmount"]}'>Maximum: ${product["giftCardAttributes"]["openAmountRange"]["openAmountMax"]}</span>
+            </p>
+         </div>
+      </div>
+  </fieldset>
+
+  <div data-giftcardtype='${product["giftCardAttributes"]["giftCardType"]}' id="giftcard_form" class="giftCardForm_root">
+      <fieldset class="giftcardProduct__fieldset">
+         <legend>
+            <span>Gift Card Information</span>
+         </legend>
+         <br>
+         <div class="giftcardProduct__field">
+            <label for="giftcard_sender_name" class="giftcardProduct__label">
+               <span>${product["giftCardAttributes"]["giftCardOptions"]["Sender Name"]["title"]}</span><span class="required">*</span>
```

Comment:
When the form fields are fixed/hard coded here we don't need to expose them in a `Map<String, GiftCardOption>` data structure but can have something like

```
 GiftCardOption getSenderNameOption()
 GiftCardOption getSenderMailOption()
... 
```